### PR TITLE
HADOOP-18050. Document Hadoop's stance on the log4jshell vulnerability.

### PR DESCRIPTION
**Pull request by jojochuang**

### CODE REVIEW

**Comment by jojochuang** [comment]
Forgot to run hugo to update the content directory. Also want to add the update in the homepage news section.

**Comment by steveloughran** [comment]
should we explicitly say 
* no ASF version of hadoop has ever been vulnerable 
* third party products may be vulnerable: consult the vendor 
* and how log4j 1.2.17 does have a vulnerability if a service is deployed, but this isnt done - and you can remove the jar (link to the cloudera jar?)
u
akso warn that 
* apps which run on hadoop may be vulneramble -check with the supplier 
* if you do find yourself replacing JARs, you MUST check and update the tar.gz files in hdfs, e.g distcp.tar.gz

if i could code, id be playing with the `hadoop findclass` command to see how it couox be used here... you should be able to use it to verify the jndi resolver is not on the classpath

has anyone tried that

**Comment by steveloughran** [comment]
hmmm, here's. 2.17 out now, still with a DoS vulnerability 


https://issues.apache.org/jira/browse/LOG4J2-3230

let's not list which versions are considered safe/unsafe

**Comment by jojochuang** [comment]
Good to know. Updated.

**Review by steveloughran — APPROVED** [review]

**Comment by steveloughran** [comment]
+1 from me

**Comment by steveloughran** [comment]
FYI this is how to programmatically check to see if a hadoop distribution has a log4j 2.x artifact on its class path and so potentially at risk from the relevant CVEs using the `hadoop findclass` command which was added a long time ago but never that documented 


First look for a class in all 2.x releases
```
> hadoop org.apache.hadoop.util.FindClass load  org.apache.logging.log4j.core.Core
Loaded org.apache.logging.log4j.core.Core as class org.apache.logging.log4j.core.Core
org.apache.logging.log4j.core.Core: file:/.../share/hadoop/common/lib/log4j-core-2.14.1.jar

> echo $0
0
```
If you find this class it means that the distribution is possibly vulnerable

now look for a class new in 2.17
```
> hadoop org.apache.hadoop.util.FindClass load  org.apache.logging.log4j.core.lookup.RuntimeStrSubstitutor

Class not found org.apache.logging.log4j.core.lookup.RuntimeStrSubstitutor
java.lang.ClassNotFoundException: Class org.apache.logging.log4j.core.lookup.RuntimeStrSubstitutor not found
        at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2565)
        at org.apache.hadoop.util.FindClass.getClass(FindClass.java:156)
        at org.apache.hadoop.util.FindClass.loadClass(FindClass.java:247)
        at org.apache.hadoop.util.FindClass.run(FindClass.java:320)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:90)
        at org.apache.hadoop.util.FindClass.main(FindClass.java:381)
  
> echo $0
3
``` 

If this class is *not* found it means the distribution is <2.17 and so is vulnerable

**Comment by Apache9** [comment]
Log4j1 has its own CVE number for this...

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-4104

I think we should also mention this in the doc?

**Comment by jojochuang** [comment]
@Apache9 updated please check
Screen Shot 2021-12-20 at 10 21 04 AM

**Comment by aajisaka** [comment]
Hi @jojochuang, would you generate the doc by the latest hugo to reduce the unnecessary formatting changes?

File: src/cve_list.md [hunk 1]
```diff
@@ -233,3 +233,19 @@ target encryption zone.
 - **Reported Date**: 2016/11/18
 - **Issue Announced**: 2017/11/08 ([general@hadoop](https://lists.apache.org/thread.html/2e16689b44bdd1976b6368c143a4017fc7159d1f2d02a5d54fe9310f@%3Cgeneral.hadoop.apache.org%3E))
 
+# Thirdparty vulnerabilities
+The following section describes thirdparty vulnerabilities that may be of interest to Hadoop users. Please contact the respective project owners for details.
+
+## [CVE-2021-44228](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228) Log4JShell Vulnerability
+
+It is understood that the log4jshell vulnerability CVE-2021-44228 impacts log4j2. Hadoop, as of 3.3.x depends on log4j 1.x, which is **NOT** susceptible to the attack. Once we migrate over to log4j2, we will adopt a version that is not susceptible to the attack, too. Therefore, no ASF version of Hadoop has ever been vulnerable. Third party products  and applications based on Hadoop *may* be vulnerable, please consult the vendor or the project owner.
```
Comment (cnauroth) [inline src/cve_list.md]: Nitpick: 2 spaces between "products" and "and".

File: content/cve_list.html [hunk 2]
```diff
@@ -350,6 +350,19 @@ <h2 id="cve-2017-3166httpcvemitreorgcgi-bincvenamecginamecve-2017-3166-apache-ha
...
 <li><strong>Issue Announced</strong>: 2017/11/08 (<a href="https://lists.apache.org/thread.html/2e16689b44bdd1976b6368c143a4017fc7159d1f2d02a5d54fe9310f@%3Cgeneral.hadoop.apache.org%3E">general@hadoop</a>)</li>
 </ul>
+<h1 id="thirdparty-vulnerabilities">Thirdparty vulnerabilities</h1>
+<p>The following section describes thirdparty vulnerabilities that may be of interest to Hadoop users. Please contact the respective project owners for details.</p>
+<h2 id="cve-2021-44228httpscvemitreorgcgi-bincvenamecginamecve-2021-44228-log4jshell-vulnerability"><a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228">CVE-2021-44228</a> Log4JShell Vulnerability</h2>
+<p>It is understood that the log4jshell vulnerability CVE-2021-44228 impacts log4j2. Hadoop, as of 3.3.x depends on log4j 1.x, which is <strong>NOT</strong> susceptible to the attack. Once we migrate over to log4j2, we will adopt a version that is not susceptible to the attack, too. Therefore, no ASF version of Hadoop has ever been vulnerable. Third party products  and applications based on Hadoop <em>may</em> be vulnerable, please consult the vendor or the project owner.</p>
```
Comment (cnauroth) [inline content/cve_list.html]: Nitpick: 2 spaces between "products" and "and".

**Review by cnauroth — APPROVED** [review]
@jojochuang , thanks so much for taking on this important documentation step.  I've only commented with some extremely minor nit-picks.  After those are resolved, +1 from me.

File: src/cve_list.md [hunk 3]
```diff
@@ -233,3 +233,19 @@ target encryption zone.
...
+
+## [CVE-2021-44228](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228) Log4JShell Vulnerability
+
+It is understood that the log4jshell vulnerability CVE-2021-44228 impacts log4j2. Hadoop, as of 3.3.x depends on log4j 1.x, which is **NOT** susceptible to the attack. Once we migrate over to log4j2, we will adopt a version that is not susceptible to the attack, too. Therefore, no ASF version of Hadoop has ever been vulnerable. Third party products and applications based on Hadoop *may* be vulnerable, please consult the vendor or the project owner.
+
+- **Versions affected**: N/A
+
+## [CVE-2021-4104](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-4104) Log4JShell Vulnerability
+
+JMSAppender in Log4j 1.2, used by all versions of Apache Hadoop, is vulnerable to the Log4JShell attack in a similar fashion to CVE-2021-4428. However, the JMSAppender is not the default configuration shipped in Hadoop. When JMSAppender is not enabled, Hadoop is not vulnerable to the attack.
```
Comment (tasanuma) [inline src/cve_list.md]:
```suggestion
JMSAppender in Log4j 1.2, used by all versions of Apache Hadoop, is vulnerable to the Log4JShell attack in a similar fashion to CVE-2021-44228. However, the JMSAppender is not the default configuration shipped in Hadoop. When JMSAppender is not enabled, Hadoop is not vulnerable to the attack.
```

Comment (jojochuang) [reply src/cve_list.md]: Done

**Review by tasanuma — APPROVED** [review]
+1 from me. Thanks for working on this, @jojochuang.

**Comment by jojochuang** [comment]
Thanks all for the feedback. Merged it. Please check back and let me know if additional information is necessary.

**Comment by tasanuma** [comment]
It seems `2021-12-17-log4jshell.html` was not generated, and I cannot access it now.
https://hadoop.apache.org/news/2021-12-17-log4jshell.html

**Comment by jojochuang** [comment]
Hi @tasanuma thanks for checking. I pushed the missing html file to the repo and I'm able to see it now.

**Comment by tasanuma** [comment]
@jojochuang I can see it now, too. Thank you!

**Pull request merged**